app/experimenter/experiments/api/v4/serializers.py

```diff
@@ -106,9 +106,31 @@ class Meta:
         fields = ("id", "arguments", "filter_expression", "enabled", "targeting")
 
     def get_filter_expression(self, obj):
+        # This needs to handle Release version numbers (80.0), Beta (80.0b1)
+        # and Nightly (80.0a1). For the algorithm used in Nimbus, any version
+        # with text on it (like 80.0a1) is less than a version without text on
+        # it (like 80.0).
+        #
+        # To account for this, we replace the ".0" in "80.0" with "!", which is
+        # the character that has the smallest ASCII value of all printable,
+        # non-whitespace characters. This will ensure that it is less than all
+        # 80-series version numbers.
+        #
+        # It isn't used now, but if a max version comparison is needed in the
+        # future the equivalent modification would be "80.*". This is because
+        # "*" is treated specially as the largest possible version in it's
+        # slot. That would allow anything that is in the 80-series, and nothing
+        # in the 81-series.
+        #
+        # To summarize, 80.! < 80.0a1 < 80.0 < 80.1 < 80.* < 81.0
+
+        firefox_min_version = obj.firefox_min_version
+        assert firefox_min_version.endswith(".0")
+        firefox_min_version = firefox_min_version[:-1] + "!"
```

Comment:
Oh you can just use this actually:

https://github.com/mozilla/experimenter/blob/main/app/experimenter/experiments/models.py#L913-L916

To get the major version and then just format it into the template

Also we haven't been in the habit of throwing asserts in the middle of application logic in general, I would avoid that pattern here.